eras/conway/impl/src/Cardano/Ledger/Conway/Governance.hs

```diff
@@ -249,7 +248,14 @@ instance EraPParams era => NoThunks (EnactState era)
 
 data RatifyState era = RatifyState
   { rsEnactState :: !(EnactState era)
-  , rsFuture :: !(StrictSeq (GovernanceActionId (EraCrypto era), GovernanceActionState era))
+  , rsFuture ::
+      !( StrictSeq
+          (GovernanceActionId (EraCrypto era), GovernanceActionState era)
+       )
+  , rsRemoved ::
+      !( StrictSeq
+          (GovernanceActionId (EraCrypto era), GovernanceActionState era)
+       )
```

Comment:
`rsRemoved` is very short-lived data. We just use it to tell the Epoch rule which actions are no longer active so we can pay back the deposits for those. We fold over the whole sequence anyways, so I think it makes sense to keep it as a sequence.

Reply:
You don't just folder over the data structure, you convert it to a list, then a Set, then a Map.
```haskell
Map.withoutKeys m . Set.fromList . toList $ fst <$> removedProposals
```

So if we keep it as a Map instead of a Seq, then we don't have to do any of these expensive conversions. It becomes:
```
m `Map.difference` removedProposals
```

Reply:
I think fold+delete would also give a similar effect

Reply:
But why not keep it as a Map? The order in which proposals are removed is irrelevant. At least we'll get the uniqueness property out of it

Reply:
Anyways. We can keep it as a sequence for now. It's not terribly important. We can change it later if necessary

Reply:
We're not keeping it as a map, `rsRemoved` is essentially built by filtering the active proposals. I feel like turning it into a map would be even more overhead. I could see a point of it being a map if we indexed into it.